c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1905,7 +2069,37 @@ void raw_delete(void* ptr) {
   caching_allocator.free(ptr);
 }
 
+std::vector<std::vector<AllocFreeEvent>> getAllocFreeEvents() {
+  return memory_tracker.get_alloc_free_events();
+}
+
+void set_mem_plan(std::vector<std::vector<AllocFreeEvent>> plan) {
```

Comment:
since you are not writing plan in this method, `plan` can be a const reference to avoid copy?